db/column_family.cc

```diff
@@ -699,19 +699,6 @@ ColumnFamilyData::~ColumnFamilyData() {
           id_, name_.c_str());
     }
   }
-
-  if (data_dirs_.size()) {  // Explicitly close data directories
-    Status s = Status::OK();
-    for (auto& data_dir_ptr : data_dirs_) {
```

Comment:
Sorry I didn't realize `ColumnFamilyData::data_dirs_` is sometimes shared. Looks like it wasn't initially in #3102, then became sometimes shared in #6358 (interestingly it was still effectively private for CFDs created via `CreateColumnFamily()`). I agree it should be shared, preferably always (including shared with `DBImpl::data_dirs_`).

I bet we'd have noticed in review if not for the use of `auto` here.